File: src/main/java/ting/leetcode/java/arrays101/NumberOfStepsToReduceANumberToZero.java
package ting.leetcode.java.arrays101;

public class NumberOfStepsToReduceANumberToZero {

    public int numberOfSteps(int num) {
        var count = 0;
        var number = num;
        while(number > 0) {
            number = (number & 1)== 1 ? number - 1 : number / 2;
            count++;
        }
        return count;
    }
}

File: src/main/kotlin/ting/leetcode/beginnerguide/NumberOfStepsToReduceANumberToZero.kt
package ting.leetcode.beginnerguide

class NumberOfStepsToReduceANumberToZero {
    fun numberOfSteps(num: Int): Int {
        var result = num
        var counter = 0
        while (result != 0) {
            result = if (result and 1 == 0) result / 2 else result - 1
            counter++
        }

        return counter
    }
}